src/mpl2/src/SACoreSoftMacro.cpp

```diff
@@ -1062,4 +1062,12 @@ void SACoreSoftMacro::calSegmentLoc(float seg_start,
   }
 }
 
+// The blockages here are only those that overlap with the annealing outline.
+void SACoreSoftMacro::addBlockages(const std::vector<Rect>& blockages)
+{
+  for (const auto& blockage : blockages) {
+    blockages_.push_back(blockage);
+  }
```

Comment:
Perhaps you could use std::vector::insert? Having a one-line function might be clearer:
`blockages_.insert(blockages_.end(), blockages .begin(), blockages.end());`